python/src/acp_sdk/server/server.py

```diff
@@ -232,7 +235,14 @@ def run(
             h11_max_incomplete_event_size,
         )
         self._server = uvicorn.Server(config)
-        self._server.run()
+
+        asyncio.get_event_loop().run_until_complete(self._run())
+
+    async def _run(self) -> None:
+        """Run the server and register the agent."""
+        server_task = asyncio.create_task(self._server.serve())
+        registration_task = asyncio.create_task(self._register_agent())
+        await asyncio.gather(server_task, registration_task)
```

Comment:
I believe this will raise exception if registration task fails which is not what we want.